newmiddleware/auth.js

```diff
@@ -0,0 +1,430 @@
+      this.verifyJwt
+    )
+    auth.delete(
+      "/v2/sites/:siteName/collections/:collectionName",
+      this.verifyJwt
+    )
+    auth.delete(
+      "/v2/sites/:siteName/collections/:collectionName/subcollections/:subcollectionName",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/collections/:collectionName/reorder",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/collections/:collectionName/subcollections/:subcollectionName/reorder",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/collections/:collectionName/move",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/collections/:collectionName/subcollections/:subcollectionName/move",
+      this.verifyJwt
+    )
+
+    // Resource Room
+    auth.get("/v2/sites/:siteName/resourceRoom", this.verifyJwt)
+    auth.post("/v2/sites/:siteName/resourceRoom", this.verifyJwt)
+    auth.get(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName",
+      this.verifyJwt
+    )
+    auth.delete(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName",
+      this.verifyJwt
+    )
+
+    // Resource Categories
+    auth.get(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName/resources/:resourceCategoryName",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName/resources",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName/resources/:resourceCategoryName",
+      this.verifyJwt
+    )
+    auth.delete(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName/resources/:resourceCategoryName",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName/resources/:resourceCategoryName/move",
+      this.verifyJwt
+    )
+
+    // Media directories
+    auth.get("/v2/sites/:siteName/media/:directoryName", this.verifyJwt)
+    auth.post("/v2/sites/:siteName/media", this.verifyJwt)
+    auth.post("/v2/sites/:siteName/media/:directoryName", this.verifyJwt)
+    auth.delete("/v2/sites/:siteName/media/:directoryName", this.verifyJwt)
+    auth.post("/v2/sites/:siteName/media/:directoryName/move", this.verifyJwt)
+
+    // Media files
+    auth.post("/v2/sites/:siteName/media/:directoryName/pages", this.verifyJwt)
+    auth.get(
+      "/v2/sites/:siteName/media/:directoryName/pages/:fileName",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/media/:directoryName/pages/:fileName",
+      this.verifyJwt
+    )
+    auth.delete(
+      "/v2/sites/:siteName/media/:directoryName/pages/:fileName",
+      this.verifyJwt
+    )
+
+    auth.use((req, res, next) => {
+      if (!req.route) {
+        return res.status(404).send("Unauthorised for unknown route")
```

Comment:
As discussed offline, since all endpoints are queried purely by our frontend, this should only be accessed if someone is attempting to ping an endpoint directly. I've updated the message sent in this case to `Unknown route requested`: https://github.com/isomerpages/isomercms-backend/pull/328/commits/0e38f6f94908110c56afe62fa657f5a01ac4482b